src/compiler/binder.ts

```diff
@@ -15,7 +15,7 @@ namespace ts {
         referenced: boolean;
     }
 
-    export function getModuleInstanceState(node: ModuleDeclaration, visited?: Map<ModuleInstanceState | undefined>): ModuleInstanceState {
+    export function getModuleInstanceState(node: ModuleDeclaration, visited?: Map<string, ModuleInstanceState | undefined>): ModuleInstanceState {

```

Comment:
Cab you make these Map<Path, ModuleInstance> and so on instead. (For all builder maps

Reply:
In this case, `visited` isn't a `Map<Path, ...>`. It stores node ids, so it actually should be a `Map<number, ...`>:

https://github.com/microsoft/TypeScript/blob/2f484289c81423bd3939ec7d5431165dc05a111b/src/compiler/binder.ts#L28-L32

Reply:
Or, rather, it probably should be a `Map<Node, ...>`, but I plan to look into cases where we use maps with `string`/`number` keys that should actually be `Node` keys at a later date.

Reply:
Sorry i was looking at builder file and wrote comment at the wrong place.. and later i thought i didnt write it so wrote again..

Reply:
I meant to write it for builder.ts (For all builder related maps/sets)